linkerd/app/outbound/src/http/peer_proxy_errors.rs

```diff
@@ -0,0 +1,186 @@
+use http::{Request, Response};
+use linkerd_app_core::{
+    errors::L5D_PROXY_ERROR,
+    svc::{layer, NewService, Service},
+};
+use linkerd_proxy_http::ClientHandle;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// PeerProxyErrors will close the accepted connection if the response is from
+/// a peer proxy and the L5D_PROXY_ERROR header is set. This means the peer
+/// proxy encountered an inbound connection error with its application and
+/// therefore the accepted connection should be torn down.
```

Comment:
very minor style nit: including the name of the documented item in the doc comment is a Go idiom; in rust we generally omit it
```suggestion
/// Closes the accepted connection if the response is from a peer
/// proxy and the L5D_PROXY_ERROR header is set. This means the
/// peer proxy encountered an inbound connection error with
/// its application and therefore the accepted connection should
/// be torn down.
```